docs/whitepaper/whitepaper.md

```diff
@@ -0,0 +1,591 @@
+
+---
+
+### 4.3.1 Ledger Connector Plugins
+
+Success is defined as:
+1. Adding support in `BIF` for a ledger invented in the future requires no `core` code changes, but instead can be implemented by simply adding a corresponding connector plugin to deal with said newly invented ledger.
+2. Client applications using the REST API and leveraging the feature checks can remain 100% functional regardless of the number and nature of deployed connector plugins in `BIF`. For example: a generic money sending application does not have to hardcode the supported ledgers it supports because the unified REST API interface (fed by the ledger connector plugins) guarantees that supported features will be operational.
+
+Because the features of different ledgers can be very diverse, the plugin interface has feature checks built into allowing callers/client applications to **determine programmatically, at runtime** if a certain feature is supported or not on a given ledger.
+
+```typescript
+export interface LedgerConnector {
+  // method to verify a signature coming from a given ledger that this connector is responsible for connecting to.
+  verifySignature(message, signature): Promise<boolean>;
+
+  // used to call methods on smart contracts or to move assets between wallets
+  transact(transactions: Transaction[]);
+
+  getPermissionScheme(): Promise<PermissionScheme>;
+
+  getTransactionFinality(): Promise<TransactionFinality>;
+
+  addForeignValidator(): Promise<void>;
+}
+
+export enum TransactionFinality {
+  GUARANTEED = "GUARANTEED",
+  NOT_GUARANTEED = "NOT_GUARANTEED
+}
+
+export enum PermissionScheme {
+  PERMISSIONED = "PERMISSIONED",
+  PERMISSIONLESS = "PERMISSIONLESS"
+}
+
+```
+
+### 4.3.2 Identity Federation Plugins
+
+Identity federation plugins operate inside the API Server and need to implement the interface of a common PassportJS Strategy:
+https://github.com/jaredhanson/passport-strategy#implement-authentication
+
+```typescript
+abstract class IdentityFederationPlugin {
+  constructor(options: any): IdentityFederationPlugin;
+  abstract authenticate(req: ExpressRequest, options: any);
+  abstract success(user, info);
+  abstract fail(challenge, status);
+  abstract redirect(url, status);
+  abstract pass();
+  abstract error(err);
+}
+```
+
+#### 4.3.1.1 X.509 Certificate Plugin
```

Comment:
What's the CA root in this case?

Reply:
Discussed in Feb 3 Contributors that not yet defined, but aspiration is to be pluggable and as distinct NPM package.

Reply:
Will add more details on CA pluggablility in a follow-up PR